.circleci/config.yml

```diff
@@ -0,0 +1,39 @@
+version: 2
+jobs:
+  build:
+    working_directory: /home/circleci/.go_workspace/src/github.com/grafana/metrictank
+    machine: true
+    steps:
+      - run: go version
+      - run: echo $GOPATH
+      - run: docker version
+      - run: docker-compose version
+      - checkout
+      - run: scripts/depends.sh
+      - run: scripts/qa/vendor.sh
+      - run: scripts/build.sh
+      - run: scripts/build_tools.sh
+      - run: scripts/build_docker.sh
+      - run: go test -v -race $(go list ./... | grep -v github.com/grafana/metrictank/chaos)
+      - run: go vet ./...
+      - run: scripts/qa/gofmt.sh
+      - run: scripts/qa/ineffassign.sh
+      - run: scripts/qa/go-generate.sh
+      - run: scripts/qa/end2end.sh
```

Comment:
end2end takes a long time, we should only run it after all other qa tests have passed.

Reply:
i've tried to take runtime into account. however some things are more important than others. let's say there's a code change that breaks end2end completely, then it's pointless to first obsess over typo's and minor code linting issues, and only find out after a few iterations of code pushing that the code is wrong and needs to be rewritten completely. I do think most of these code health qa scripts can be run in parallel which is something i want to migrate towards (at some point)

Reply:
I think that is backwards.  I think the linting issues and typo's should be resolved first.

By your logic, you will have to wait 20minutes between each code iteration before being told you have a simple typo. It seems insane to be running a full end2end test mulitple times as you fix minor issues that could be reported in a few seconds.  With your approach it will take 10x longer to get the same amount of work done.

Reply:
good point. will fix